network_client.go

```diff
@@ -51,9 +52,21 @@ func (nc *NetworkClient) send(topic string, partition int32, batch []*ProducerRe
 	}
 	responseChan := nc.selector.Send(leader, request)
 
-	if nc.requiredAcks > 0 {
-		go listenForResponse(topic, partition, batch, responseChan)
-	} else {
+	go nc.listenForResponse(topic, partition, batch, responseChan)
+}
+
+func (nc *NetworkClient) listenForResponse(topic string, partition int32, batch []*ProducerRecord, responseChan <-chan *rawResponseAndError) {
+	response := <-responseChan
```

Comment:
I removed if blocks with `acks == 0` in selector.go as well, so that network_client can get at least send errors and refresh its cached leaders.
https://github.com/elodina/siesta-producer/pull/5/files?diff=unified#diff-186c7b73f4ec5bbf632bfed92b347e75L119